plugin doesn't work on Windows

While testing something on `unplugin-icons` I just open the plugin to see it, but then: 

![imagen](https://user-images.githubusercontent.com/6311119/133635062-229fae8f-73df-4c8f-8f50-4f41ea176303.png)

The `fetch` api is the originator of the error...

@antfu cannot compile the project, I'm trying to fix it, seems you haven't install `vue` dep: ```shell
> vite-plugin-inspect@0.3.6 build:js F:\work\projects\quini\GitHub\antfu\vite-plugin-inspect
> tsup src/node/index.ts --format cjs,esm --dts --env.NODE_ENV production

 CLI  Building entry: src/node/index.ts
 CLI  Using tsconfig: F:\work\projects\quini\GitHub\antfu\vite-plugin-inspect\tsconfig.json
 CLI  tsup v4.14.0
 CLI  Target: node12
 CJS  Build start
 ESM  Build start
 > node_modules/.pnpm/vue-router@4.0.11/node_modules/vue-router/dist/vue-router.cjs.js:10:18: error: Could not resolve "vue" (mark it as external to exclude it from the bundle, or surround it with try/catch to h
andle the failure at run-time)
    10 │ var vue = require('vue');
       ╵                   ~~~~~

 > node_modules/.pnpm/vue-router@4.0.11/node_modules/vue-router/dist/vue-router.cjs.js:10:18: error: Could not resolve "vue" (mark it as external to exclude it from the bundle, or surround it with try/catch to h
andle the failure at run-time)
    10 │ var vue = require('vue');
       ╵                   ~~~~~

 ESM  Build failed
Error: Build failed with 1 error:
node_modules/.pnpm/vue-router@4.0.11/node_modules/vue-router/dist/vue-router.cjs.js:10:18: error: Could not resolve "vue" (mark it as external to exclude it from the bundle, or surround it with try/catch to hand
le the failure at run-time)
    at failureErrorWithLog (F:\work\projects\quini\GitHub\antfu\vite-plugin-inspect\node_modules\.pnpm\esbuild@0.12.17\node_modules\esbuild\lib\main.js:1449:15)
    at F:\work\projects\quini\GitHub\antfu\vite-plugin-inspect\node_modules\.pnpm\esbuild@0.12.17\node_modules\esbuild\lib\main.js:1131:28
    at runOnEndCallbacks (F:\work\projects\quini\GitHub\antfu\vite-plugin-inspect\node_modules\.pnpm\esbuild@0.12.17\node_modules\esbuild\lib\main.js:921:63)
    at buildResponseToResult (F:\work\projects\quini\GitHub\antfu\vite-plugin-inspect\node_modules\.pnpm\esbuild@0.12.17\node_modules\esbuild\lib\main.js:1129:7)
    at F:\work\projects\quini\GitHub\antfu\vite-plugin-inspect\node_modules\.pnpm\esbuild@0.12.17\node_modules\esbuild\lib\main.js:1236:14
    at F:\work\projects\quini\GitHub\antfu\vite-plugin-inspect\node_modules\.pnpm\esbuild@0.12.17\node_modules\esbuild\lib\main.js:609:9
    at handleIncomingPacket (F:\work\projects\quini\GitHub\antfu\vite-plugin-inspect\node_modules\.pnpm\esbuild@0.12.17\node_modules\esbuild\lib\main.js:706:9)
    at Socket.readFromStdout (F:\work\projects\quini\GitHub\antfu\vite-plugin-inspect\node_modules\.pnpm\esbuild@0.12.17\node_modules\esbuild\lib\main.js:576:7)
    at Socket.emit (node:events:378:20)
    at addChunk (node:internal/streams/readable:313:12)
 CJS  Build failed
 ERROR  Command failed with exit code 1.
ERROR: "build:js" exited with 1.
 ERROR  Command failed with exit code 1.
```

--- a/src/node/index.ts
+++ b/src/node/index.ts
@@ -10,6 +10,10 @@ import { ModuleInfo, TransformInfo } from '../types'
 
 const debug = _debug('vite-plugin-inspect')
 
+const isWindows = process.platform === 'win32'
+const windowsRe = /^([A-Z]+):\/(.*)/
+const WindowResolveIdPrefix = '/__resolve_windows/'
+
 export interface Options {
   /**
    * Enable the inspect plugin (could be some performance overhead)
@@ -45,6 +49,7 @@ function PluginInspect(options: Options = {}): Plugin {
 
   const transformMap: Record<string, TransformInfo[]> = {}
   const idMap: Record<string, string> = {}
+  const windowsIdMap: Record<string, string> = {}
 
   function hijackPlugin(plugin: Plugin) {
     if (plugin.transform) {
@@ -52,9 +57,18 @@ function PluginInspect(options: Options = {}): Plugin {
       const _transform = plugin.transform
       plugin.transform = async function(this: any, ...args: any[]) {
         const code = args[0]
-        const id = args[1]
+        let id = args[0]
+        let _result
         const start = Date.now()
-        const _result = await _transform.apply(this, args as any)
+        if (isWindows && id && id.startsWith(WindowResolveIdPrefix)) {
+          id = windowsIdMap[id]
+          args = new Array<any>(id)
+          args.unshift(...args.splice(1))
+          _result = await _transform.apply(this, args as any)
+        }
+        else {
+          _result = await _transform.apply(this, args as any)
+        }
         const end = Date.now()
 
         const result = typeof _result === 'string' ? _result : _result?.code
@@ -78,7 +92,10 @@ function PluginInspect(options: Options = {}): Plugin {
       debug('hijack plugin load', plugin.name)
       const _load = plugin.load
       plugin.load = async function(this: any, ...args: any[]) {
-        const id = args[0]
+        let id = args[0]
+        if (isWindows && id && id.startsWith(WindowResolveIdPrefix))
+          id = windowsIdMap[id]
+
         const start = Date.now()
         const _result = await _load.apply(this, args as any)
         const end = Date.now()
@@ -96,8 +113,17 @@ function PluginInspect(options: Options = {}): Plugin {
       debug('hijack plugin resolveId', plugin.name)
       const _resolveId = plugin.resolveId
       plugin.resolveId = async function(this: any, ...args: any[]) {
-        const id = args[0]
-        const _result = await _resolveId.apply(this, args as any)
+        let id = args[0]
+        let _result
+        if (isWindows && id && id.startsWith(WindowResolveIdPrefix)) {
+          id = windowsIdMap[id]
+          args = new Array<any>(id)
+          args.unshift(...args.splice(1))
+          _result = await _resolveId.apply(this, args as any)
+        }
+        else {
+          _result = await _resolveId.apply(this, args as any)
+        }
 
         const result = typeof _result === 'object' ? _result?.id : _result
 
@@ -177,8 +203,16 @@ function PluginInspect(options: Options = {}): Plugin {
         res.end()
       }
       else if (pathname === '/resolve') {
-        const id = parseQuery(search).id as string
-        res.write(JSON.stringify({ id: resolveId(id) }, null, 2))
+        let id = resolveId(parseQuery(search).id as string)
+        if (isWindows) {
+          const match = id.match(windowsRe)
+          if (match) {
+            const original = id
+            id = `${WindowResolveIdPrefix}${match[1]}/${match[2]}`
+            windowsIdMap[id] = original
+          }
+        }
+        res.write(JSON.stringify({ id }, null, 2))
         res.end()
       }
       else if (pathname === '/clear') {

--- a/src/node/index.ts
+++ b/src/node/index.ts
@@ -52,23 +52,28 @@ function PluginInspect(options: Options = {}): Plugin {
   const windowsIdMap: Record<string, string> = {}
 
   function hijackPlugin(plugin: Plugin) {
+    function resolveArguments(...args: any[]) {
+      let id = args[0]
+      if (isWindows && id && id.startsWith(WindowResolveIdPrefix)) {
+        id = windowsIdMap[id]
+        args = new Array<any>(...args.splice(1))
+        args.unshift(id)
+      }
+      return args
+    }
     if (plugin.transform) {
       debug('hijack plugin transform', plugin.name)
       const _transform = plugin.transform
       plugin.transform = async function(this: any, ...args: any[]) {
         const code = args[0]
-        let id = args[0]
-        let _result
-        const start = Date.now()
+        let id = args[1]
         if (isWindows && id && id.startsWith(WindowResolveIdPrefix)) {
           id = windowsIdMap[id]
-          args = new Array<any>(...args.splice(1))
-          args.unshift(id)
-          _result = await _transform.apply(this, args as any)
+          args = new Array<any>(...args.splice(2))
+          args.unshift(code, id)
         }
-        else {
-          _result = await _transform.apply(this, args as any)
-        }
+        const start = Date.now()
+        const _result = await _transform.apply(this, args as any)
         const end = Date.now()
 
         const result = typeof _result === 'string' ? _result : _result?.code
@@ -92,18 +97,11 @@ function PluginInspect(options: Options = {}): Plugin {
       debug('hijack plugin load', plugin.name)
       const _load = plugin.load
       plugin.load = async function(this: any, ...args: any[]) {
-        let id = args[0]
-        let _result
+        args = resolveArguments(args)
+        const id = args[0]
+
         const start = Date.now()
-        if (isWindows && id && id.startsWith(WindowResolveIdPrefix)) {
-          id = windowsIdMap[id]
-          args = new Array<any>(...args.splice(1))
-          args.unshift(id)
-          _result = await _load.apply(this, args as any)
-        }
-        else {
-          _result = await _load.apply(this, args as any)
-        }
+        const _result = await _load.apply(this, args as any)
         const end = Date.now()
 
         const result = typeof _result === 'string' ? _result : _result?.code
@@ -119,17 +117,10 @@ function PluginInspect(options: Options = {}): Plugin {
       debug('hijack plugin resolveId', plugin.name)
       const _resolveId = plugin.resolveId
       plugin.resolveId = async function(this: any, ...args: any[]) {
-        let id = args[0]
-        let _result
-        if (isWindows && id && id.startsWith(WindowResolveIdPrefix)) {
-          id = windowsIdMap[id]
-          args = new Array<any>(...args.splice(1))
-          args.unshift(id)
-          _result = await _resolveId.apply(this, args as any)
-        }
-        else {
-          _result = await _resolveId.apply(this, args as any)
-        }
+        args = resolveArguments(args)
+        const id = args[0]
+
+        const _result = await _resolveId.apply(this, args as any)
 
         const result = typeof _result === 'object' ? _result?.id : _result
 

--- a/src/node/index.ts
+++ b/src/node/index.ts
@@ -53,11 +53,10 @@ function PluginInspect(options: Options = {}): Plugin {
 
   function hijackPlugin(plugin: Plugin) {
     function resolveArguments(...args: any[]) {
-      let id = args[0]
-      if (isWindows && id && id.startsWith(WindowResolveIdPrefix)) {
-        id = windowsIdMap[id]
-        args = new Array<any>(id, ...args.splice(1))
-      }
+      const id = args[0]
+      if (isWindows && id && id.startsWith(WindowResolveIdPrefix))
+        args[0] = windowsIdMap[id]
+
       return args
     }
     if (plugin.transform) {
@@ -68,7 +67,7 @@ function PluginInspect(options: Options = {}): Plugin {
         let id = args[1]
         if (isWindows && id && id.startsWith(WindowResolveIdPrefix)) {
           id = windowsIdMap[id]
-          args = new Array<any>(code, id, ...args.splice(2))
+          args[1] = id
         }
 
         const start = Date.now()

--- a/src/node/index.ts
+++ b/src/node/index.ts
@@ -56,8 +56,7 @@ function PluginInspect(options: Options = {}): Plugin {
       let id = args[0]
       if (isWindows && id && id.startsWith(WindowResolveIdPrefix)) {
         id = windowsIdMap[id]
-        args = new Array<any>(...args.splice(1))
-        args.unshift(id)
+        args = new Array<any>(id, ...args.splice(1))
       }
       return args
     }
@@ -69,8 +68,7 @@ function PluginInspect(options: Options = {}): Plugin {
         let id = args[1]
         if (isWindows && id && id.startsWith(WindowResolveIdPrefix)) {
           id = windowsIdMap[id]
-          args = new Array<any>(...args.splice(2))
-          args.unshift(code, id)
+          args = new Array<any>(code, id, ...args.splice(2))
         }
 
         const start = Date.now()

--- a/src/node/index.ts
+++ b/src/node/index.ts
@@ -62,8 +62,8 @@ function PluginInspect(options: Options = {}): Plugin {
         const start = Date.now()
         if (isWindows && id && id.startsWith(WindowResolveIdPrefix)) {
           id = windowsIdMap[id]
-          args = new Array<any>(id)
-          args.unshift(...args.splice(1))
+          args = new Array<any>(...args.splice(1))
+          args.unshift(id)
           _result = await _transform.apply(this, args as any)
         }
         else {
@@ -93,11 +93,17 @@ function PluginInspect(options: Options = {}): Plugin {
       const _load = plugin.load
       plugin.load = async function(this: any, ...args: any[]) {
         let id = args[0]
-        if (isWindows && id && id.startsWith(WindowResolveIdPrefix))
-          id = windowsIdMap[id]
-
+        let _result
         const start = Date.now()
-        const _result = await _load.apply(this, args as any)
+        if (isWindows && id && id.startsWith(WindowResolveIdPrefix)) {
+          id = windowsIdMap[id]
+          args = new Array<any>(...args.splice(1))
+          args.unshift(id)
+          _result = await _load.apply(this, args as any)
+        }
+        else {
+          _result = await _load.apply(this, args as any)
+        }
         const end = Date.now()
 
         const result = typeof _result === 'string' ? _result : _result?.code
@@ -117,8 +123,8 @@ function PluginInspect(options: Options = {}): Plugin {
         let _result
         if (isWindows && id && id.startsWith(WindowResolveIdPrefix)) {
           id = windowsIdMap[id]
-          args = new Array<any>(id)
-          args.unshift(...args.splice(1))
+          args = new Array<any>(...args.splice(1))
+          args.unshift(id)
           _result = await _resolveId.apply(this, args as any)
         }
         else {

--- a/src/node/index.ts
+++ b/src/node/index.ts
@@ -66,8 +66,12 @@ function PluginInspect(options: Options = {}): Plugin {
       const _transform = plugin.transform
       plugin.transform = async function(this: any, ...args: any[]) {
         const code = args[0]
-        args = resolveArguments(args.splice(1))
-        const id = args[1]
+        let id = args[1]
+        if (isWindows && id && id.startsWith(WindowResolveIdPrefix)) {
+          id = windowsIdMap[id]
+          args = new Array<any>(...args.splice(2))
+          args.unshift(code, id)
+        }
 
         const start = Date.now()
         const _result = await _transform.apply(this, args as any)

--- a/src/node/index.ts
+++ b/src/node/index.ts
@@ -66,12 +66,9 @@ function PluginInspect(options: Options = {}): Plugin {
       const _transform = plugin.transform
       plugin.transform = async function(this: any, ...args: any[]) {
         const code = args[0]
-        let id = args[1]
-        if (isWindows && id && id.startsWith(WindowResolveIdPrefix)) {
-          id = windowsIdMap[id]
-          args = new Array<any>(...args.splice(2))
-          args.unshift(code, id)
-        }
+        args = resolveArguments(args.splice(1))
+        const id = args[1]
+
         const start = Date.now()
         const _result = await _transform.apply(this, args as any)
         const end = Date.now()

